pkg/prometheus/promcfg_test.go

```diff
@@ -7374,14 +7375,51 @@ func TestScrapeConfigSpecConfigWithDigitalOceanSD(t *testing.T) {
 						},
 						FollowRedirects: ptr.To(true),
 						EnableHTTP2:     ptr.To(true),
-						Port:            ptr.To(9100),
+						Port:            ptr.To(int32(9100)),
 						RefreshInterval: ptr.To(monitoringv1.Duration("30s")),
 					},
 				},
 			},
-			golden: "ScrapeConfigSpecConfig_DigitalOceanSD.golden",
-		},
-		{
+			version: "2.40.0",
+			golden:  "ScrapeConfigSpecConfig_DigitalOceanSD.golden",
+		}, {
+			name: "digitalocean_sd_config_with_unsupported_version",
```

Comment:
Not sure we want to test for cases which won't go through the resource selector in the first place. I would suggest removing this test case, the one that you added at `resource_selector_test.go` should be sufficient.